lib/Doctrine/Migrations/MigrationRepository.php

```diff
@@ -391,15 +391,23 @@ public function getNextVersion() : ?string
         return $this->getRelativeVersion($this->getCurrentVersion(), 1);
     }
 
-    private function loadMigrationsFromDirectory() : void
+    private function loadMigrationsFromDirectories() : void
     {
-        $migrationsDirectory = $this->configuration->getMigrationsDirectory();
+        $migrationDirectories = $this->configuration->getMigrationDirectories();
 
-        if (count($this->versions) !== 0 || $migrationsDirectory === null) {
+        if (count($this->versions) !== 0 || count($migrationDirectories) === 0) {
             return;
         }
 
-        $this->registerMigrationsFromDirectory($migrationsDirectory);
+        foreach ($migrationDirectories as $namespace => $paths) {
```

Comment:
I think that exposes the internals of `Configuration` a bit too much, maybe we can provide a method there to encapsulate things in a better way.  Like:

```php
class Configuration
{
    /**
     * @return iterable<array<string|null>>
     */
    public function iterateMigrationDirectories() : iterable
    {
        foreach ($this->migrationsDirectories as $namespace => $paths) {
            foreach ($paths as $path) {
                yield [$path, $namespace ?: null];
            }
        }
    }
}
```

Then we could do this here:

```php
foreach ($this->configuration->iterateMigrationDirectories() as $directoryConfiguration) {
    $this->registerMigrations(
        $this->migrationFinder->findMigrations(...$directoryConfiguration)
    );
}
```

We would also be able to drop `count($migrationDirectories) === 0` too since we would have an empty iterator when nothing is configured.

Reply:
If `Configuration#getMigrationDirectories()` is only used by this method we can simply no create it too :smile: